crates/steel-core/src/primitives/strings.rs

```diff
@@ -221,24 +221,46 @@ pub fn string_ci_equals(left: &SteelString, right: &SteelString) -> bool {
 
 #[function(name = "string-ref", constant = true)]
 pub fn string_ref(value: &SteelString, index: usize) -> Result<SteelVal> {
-    if index >= value.len() {
-        stop!(Generic => "string-ref: index out of bounds: index: {}, string length: {}", index, value);
-    }
+    let res = if index < value.len() {
+        value.chars().nth(index)
+    } else {
+        None
+    };
 
-    Ok(SteelVal::CharV(value.as_str().chars().nth(index).unwrap()))
+    if let Some(ch) = res {
+        Ok(SteelVal::CharV(ch))
+    } else {
+        stop!(Generic => "string-ref: index out of bounds: index: {}, string length: {}", index, value.len());
+    }
 }
 
 #[function(name = "substring", constant = true)]
 pub fn substring(value: &SteelString, i: usize, j: usize) -> Result<SteelVal> {
```

Comment:
:question: Should `j` be optional? (I think it is in other Schemes)

Reply:
Yeah it looks like it is optional in Racket - did you want to make that change here?